js/app/modules/contentGroup/contentGroup.js

```diff
@@ -267,6 +271,10 @@ const ContentGroup = new Module.Class({
 
     },
 
+    _on_reading_history_changed: function () {
+        this._arrangement.mark_read(ReadingHistoryModel.get_default().get_read_articles());
```

Comment:
Our models have always had all `CONSTRUCT_ONLY` properties - i.e. been immutable. My idea was to stick with that and have mark_read operate similar to how highlight operates